File: android/src/main/kotlin/com/fit321/fitui/tokens/FitSpacing.kt
package com.fit321.fitui.tokens

import androidx.compose.ui.unit.dp

/**
 * Spacing tokens — Tailwind half-step scale (2 / 4 / 6 / 8 / 10 / 12 /
 * 14 / 16 / 18 / 20 / 24 / 28 / 32 / 40 / 48).
 */
object FitSpacing {
    val sp0   = 0.dp
    val sp0_5 = 2.dp
    val sp1   = 4.dp
    val sp1_5 = 6.dp
    val sp2   = 8.dp
    val sp2_5 = 10.dp
    val sp3   = 12.dp
    val sp3_5 = 14.dp
    val sp4   = 16.dp
    val sp4_5 = 18.dp
    val sp5   = 20.dp
    val sp6   = 24.dp
    val sp7   = 28.dp
    val sp8   = 32.dp
    val sp9   = 40.dp
    val sp10  = 48.dp
}

/**
 * Border-radius tokens — extended scale (4 / 6 / 8 / 10 / 12 / 14 /
 * 16 / 20 / 28 / full).
 */
object FitRadius {
    val r4   = 4.dp
    val r6   = 6.dp
    val r8   = 8.dp
    val r10  = 10.dp
    val r12  = 12.dp
    val r14  = 14.dp
    val r16  = 16.dp
    val r20  = 20.dp
    val r28  = 28.dp
    val full = 9999.dp

    // Semantic aliases
    val xs   = r4
    val sm   = r8
    val md   = r12
    val lg   = r16

    // Component-specific
    val button       = 99.dp
    val input        = r12
    val card         = r16
    val sheet        = r16
    val settingsCard = r16
    val badge        = r6
    val selectRow    = r8
}

/**
 * Component sizes — avatar / button / navbar / icons / etc.
 */
object FitSize {
    // Button (Apple HIG 50dp)
    val buttonLgHeight = 50.dp
    val buttonMdHeight = 44.dp
    val buttonSmHeight = 40.dp

    val tapMin = 44.dp

    // Input
    val inputHeight = 56.dp

    // Avatar (5 semantic sizes)
    val avatarXs = 24.dp
    val avatarSm = 32.dp
    val avatarMd = 40.dp
    val avatarLg = 48.dp
    val avatarXl = 80.dp

    // Navbar
    val navbarHeight = 56.dp
    val navbarItemSize = 56.dp

    // Icons
    val iconSm = 14.dp
    val iconMd = 16.dp
    val iconLg = 24.dp

    val iconBtnSize = 32.dp

    // Settings card
    val settingsCardIcon = 24.dp
    val settingsCardChevron = 16.dp

    // Select row
    val selectRowCheck = 22.dp
    val selectRowDot = 10.dp

    // Checkbox
    val checkboxSize = 28.dp

    // Toggle
    val toggleWidth = 48.dp
    val toggleHeight = 28.dp
    val toggleThumb = 22.dp

    // Stepper
    val stepperHeight = 48.dp
    val stepperButton = 48.dp

    // Toast/Snackbar
    val toastIcon = 16.dp
    val snackbarDot = 6.dp

    // Password eye
    val eyeIcon = 18.dp
}

File: Sources/FitUI/Tokens/FitSpacing.swift
import SwiftUI

// MARK: - 321Fit Spacing & Radius
// Source: design-tokens/tokens/spacing.json + components.json

public enum FitSpacing {
    public static let sp0: CGFloat = 0
    public static let sp1: CGFloat = 4
    public static let sp2: CGFloat = 8
    public static let sp3: CGFloat = 12
    public static let sp4: CGFloat = 16
    public static let sp5: CGFloat = 20
    public static let sp6: CGFloat = 24
    public static let sp7: CGFloat = 32
    public static let sp8: CGFloat = 40
    public static let sp9: CGFloat = 48
}

public enum FitRadius {
    public static let xs: CGFloat = 4
    public static let sm: CGFloat = 8
    public static let md: CGFloat = 12
    public static let lg: CGFloat = 16
    public static let full: CGFloat = 9999

    // Component-specific
    public static let button: CGFloat = 99
    public static let input: CGFloat = 12
    public static let card: CGFloat = 16
    public static let sheet: CGFloat = 16
    public static let settingsCard: CGFloat = 16
    public static let badge: CGFloat = 6
    public static let selectRow: CGFloat = 8
}

public enum FitSize {
    // Button (per feedback_spacing_typography — Apple HIG 50pt)
    public static let buttonLgHeight: CGFloat = 50   // footer primary CTA
    public static let buttonMdHeight: CGFloat = 44   // card-level CTA, tap-target min
    public static let buttonSmHeight: CGFloat = 40   // compact in sheets / cards
    public static let buttonHeight: CGFloat = buttonLgHeight   // legacy alias

    public static let tapMin: CGFloat = 44            // iOS HIG minimum tap target

    // Input
    public static let inputHeight: CGFloat = 56

    // Avatar (5 semantic sizes per spacing.json)
    public static let avatarXs: CGFloat = 24
    public static let avatarSm: CGFloat = 32
    public static let avatarMd: CGFloat = 40          // standard list avatar
    public static let avatarLg: CGFloat = 48          // event sheet / card avatar
    public static let avatarXl: CGFloat = 80          // profile hero

    // Navbar
    public static let navbarHeight: CGFloat = 56
    public static let navbarItemSize: CGFloat = 56

    // Icons
    public static let iconSm: CGFloat = 14
    public static let iconMd: CGFloat = 16
    public static let iconLg: CGFloat = 24

    // Header / toolbar
    public static let iconBtnSize: CGFloat = 32

    // Settings card
    public static let settingsCardIcon: CGFloat = 24
    public static let settingsCardChevron: CGFloat = 16

    // Select row
    public static let selectRowCheck: CGFloat = 22
    public static let selectRowDot: CGFloat = 10

    // Checkbox
    public static let checkboxSize: CGFloat = 28

    // Toggle
    public static let toggleWidth: CGFloat = 48
    public static let toggleHeight: CGFloat = 28
    public static let toggleThumb: CGFloat = 22

    // Stepper
    public static let stepperHeight: CGFloat = 48
    public static let stepperButton: CGFloat = 48

    // Toast/Snackbar
    public static let toastIcon: CGFloat = 16
    public static let snackbarDot: CGFloat = 6

    // Password eye
    public static let eyeIcon: CGFloat = 18
}
